plutus-playground-server/usecases/GameSimulations.hs

```diff
@@ -0,0 +1,95 @@
+{-# LANGUAGE NamedFieldPuns    #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeOperators     #-}
+{-# OPTIONS_GHC -Wno-missing-import-lists #-}
+
+module GameSimulations where
+
+import           Game                  (GuessParams (..), LockParams (..))
+import qualified Ledger.Ada            as Ada
+import           Playground.Types      (ContractCall (AddBlocks, CallEndpoint), EndpointName,
+                                        FunctionSchema (FunctionSchema), PayToWalletParams (..),
+                                        Simulation (Simulation), SimulatorAction, SimulatorWallet (SimulatorWallet),
+                                        argumentValues, arguments, blocks, caller, endpointName, simulationActions,
+                                        simulationName, simulationWallets, simulatorWalletBalance,
+                                        simulatorWalletWallet)
+import           Schema                (ToArgument, toArgument)
+import           Wallet.Emulator.Types (Wallet (Wallet), getWallet)
+
+gameSimulations :: [Simulation]
+gameSimulations = [basicGame, justWait, badGuess, fundTransfer]
+  where
+    wallet1 = Wallet {getWallet = 1}
+    wallet2 = Wallet {getWallet = 2}
+    wallet3 = Wallet {getWallet = 3}
+    basicGame =
+        Simulation
+            { simulationName = "Basic Game"
+            , simulationWallets =
+                  (\wallet ->
+                       SimulatorWallet
+                           { simulatorWalletWallet = wallet
+                           , simulatorWalletBalance = Ada.lovelaceValueOf 10000
+                           }) <$>
+                  [wallet1, wallet2]
+            , simulationActions =
+                  [ call wallet1 "lock" $
+                    LockParams
+                        {secretWord = "Plutus", amount = Ada.lovelaceOf 100}
+                  , call wallet2 "guess" (GuessParams {guessWord = "Plutus"})
+                  ]
+            }
+    justWait =
+        Simulation
+            { simulationName = "Just Wait"
+            , simulationWallets =
+                  (\wallet ->
+                       SimulatorWallet
+                           { simulatorWalletWallet = wallet
+                           , simulatorWalletBalance = Ada.lovelaceValueOf 100
+                           }) <$>
+                  [wallet1, wallet2]
```

Comment:
again, it seems like `simulationWallets = simulatorWallet 1000 <$> [wallet1, wallet2]` would make the test easier to read.

Reply:
Yeah, much better. Thanks.